src/index.js

```diff
@@ -9,13 +10,27 @@ import Node from "./utils/Node";
 const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(<App />);
 
-// 현재 작성중인 로직 확인용(정상배포 전 삭제 필요)
+// NOTE: 현재 작성중인 로직 확인용(정상배포 전 삭제 필요)
 const fiberRootNode = deepCopy(root._internalRoot);
 const fiberTree = new Node();
 
+const getCircularReplacer = () => {
+  const seen = new WeakSet();
+
+  return (key, value) => {
+    if (typeof value === "object" && value !== null) {
+      if (seen.has(value)) return;
+
+      seen.add(value);
+    }
+
+    return value;
+  };
+};
+
```

Comment:
```suggestion
const seen = new WeakSet();
const getCircularReplacer = (key, value) => {
  if (key === "elementType" && typeof value === "function") return { name : value.name };

  if (typeof value === "object" && value !== null) {
    if (seen.has(value)) return;

    seen.add(value);
  }

  return value;
};
```

JSON.stringify에서 순환참조 property를 삭제하는 함수 실행 과정에서 데이터 손실이 있습니다. 
그래서 elementType 속성값이 예상대로 나오지 않아 조건문 `if (key === "elementType" && typeof value === "function") return { name : value.name };`을 추가하는 것이 좋은 것 같습니다.

그리고 코드를 조금 더 짧게 수정해보았습니다. 
`getCircularReplacer`함수는 한 번만 실행될 것이기 때문에, 함수 내부에 있던 변수`seen`을 전역에서 선언해도 이전과 비교했을 때 성능 차이가 없다고 생각했습니다.